File: sumofNum.py

sum = 6

arr = [1,3,5,7,1]

for i,item in enumerate(arr):
    for j,value in enumerate(arr, start=i):
        if item+value == 6:
            if i != j:
                print('Match' + str(item) + ' ' + str(value))

File: sumofNum.php
<?php

$sum = 6;

$arr = [1,3,5,7,1];

for($i = 0; $i < count($arr); $i++) {
    for($j = $i+1; $j < count($arr);$j++) {
        $currentSum = $arr[$i] + $arr[$j];
        if($currentSum == 6) {
                echo ' i = ' . $i . ' j = ' . $j;
                echo '<br>';
                echo $arr[$i] . ' ' . $arr[$j];
                echo '<br>';
        }
    }
}